### ISSUE TITLE
Datastore Emulator /shutdown results in ECONNRESET in Node.js

### ISSUE THREAD

**Issue by z-vr** [GoogleCloudPlatform/google-cloud-datastore#188]
I'm trying to shutdown _Datastore Emulator_ from Node.js:

```js
const options = {
    host: 'localhost',
    port: 8081,
    path: '/shutdown',
    method: 'POST',
};
http.request(options, ({ statusCode }) => {
    console.log(statusCode)
})
    .on('error', (err) => {
        console.log(err);
    })
    .end();
```

But most of the times I get 

```
{ Error: socket hang up
    at createHangUpError (_http_client.js:329:15)
    at Socket.socketOnEnd (_http_client.js:421:23)
    at emitNone (events.js:110:20)
    at Socket.emit (events.js:207:7)
    at endReadableNT (_stream_readable.js:1057:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9) code: 'ECONNRESET' }
```
which means that the emulator server closed the http connection before completing it properly. Rarely, I would get successful result with `200` status code -- which should be happening every time.

**Comment by wsh** [comment]
Hi there! This repo contains the low-level Java and Python Cloud Datastore libraries. It's not the right place for issues regarding the emulator. [Take a look here](https://cloud.google.com/datastore/docs/community) for better support options.

**Comment by z-vr** [comment]
@wsh I'm just trying to be helpful in reporting a bug with Datastore Emulator because there is no other place but if nobody cares then me neither.

**Comment by pcostell** [comment]
Thanks for the bug report @z-vr . This was an oversight on our part -- the emulator is supposed to wait after sending the `OK` on `/shutdown` before killing itself, but it was not. This should be fixed in the next emulator release (will not happen until January).

**Comment by z-vr** [comment]
@pcostell no worries, cheers for looking into it!

**Comment by wsh** [comment]
@z-vr hey, to be clear, we do appreciate the bug report--I just wanted to make sure it was followed up on. (We need to get our ducks in a row with respect to issue reporting.) @pcostell fixed this for emulator release 1.4.1, which should land this week or next in your SDK. Thanks again for letting us know.
